### DISCUSSION TITLE
sk-ssh-ed25519@openssh.com support (yubikey)

### DISCUSSION THREAD

**Discussion by ghost** [FiloSottile/age#296] [General]
Hello,

far as i know age don't support yubikey, but wouldn't it be easy to implement ed25519-sk to add semi yubikey support?
Every encryption need press of yubikey.

`Warning: recipients file "age.pub": ignoring unsupported SSH key of type "sk-ssh-ed25519@openssh.com" at line 1`

**Comment by FiloSottile** [comment]
Unfortunately no, this is just not possible, since FIDO2 doesn't support decryption operations. We can add **symmetric** support for FIDO2 security keys, and might as a plugin, but not asymmetric support like what's needed to use SSH keys.

**Reply by ghost** [reply]
okay thank you for the answer.

**Comment by str4d** [comment]
As an aside, the age protocol _does_ support YubiKey (in asymmetric mode) via a plugin. You can find the beta release here: https://github.com/str4d/age-plugin-yubikey

`age` hasn't yet implemented plugin support, but you can test the plugin with `rage`: https://github.com/str4d/rage
